### PR TITLE
[Tests] refactor test paths

### PR DESCRIPTION
**Pull request by mertcanaltin**
- Edited fs.readdirSync call
- Operations done within the loop of the files object are set as function
- This function is called in a loop by returning the keys of the files object using the Object.keys function 🚀

### CODE REVIEW

**Comment by ljharb** [comment]
why is this an improvement?

**Comment by mertcanaltin** [comment]
> bu neden bir gelişme?

clean use :/

**Comment by ljharb** [comment]
What specifically about your approach is cleaner?

**Comment by mertcanaltin** [comment]
> Yaklaşımınız hakkında özellikle ne daha temiz?

Defining base and files objects can be done in one line as follows
`const files = {
  base: require('../base'),
};
 `

When calling fs.readdirSync, it can be changed as follows to merge the folder path with path.join and verify that the directory is rules: ı think

`const rulesDir = path.join(__dirname, '../rules');
fs.readdirSync(rulesDir).forEach((name) => {
  if (name === 'react.js' || name === 'react-a11y.js') {
    return;
  }

  // eslint-disable-next-line import/no-dynamic-require
  files[name] = require(path.join(rulesDir, name));
});`

Operations done inside the loop of the files object can be passed to a function like this

Operations done within the loop of the files object can be transferred to the function

I thought it would be healthier for it to be more organized and understandable, my friend.

File: packages/eslint-config-airbnb/test/test-base.js [hunk 1]
```diff
@@ -2,22 +2,21 @@ import fs from 'fs';
...
-fs.readdirSync(path.join(__dirname, '../rules')).forEach((name) => {
+const rulesDir = path.join(__dirname, '../rules');
+fs.readdirSync(rulesDir).forEach((name) => {
   if (name === 'react.js' || name === 'react-a11y.js') {
     return;
   }
 
   // eslint-disable-next-line import/no-dynamic-require
-  files[name] = require(`../rules/${name}`); // eslint-disable-line global-require
+  files[name] = require(path.join(rulesDir, name));
```
Comment (ljharb) [inline packages/eslint-config-airbnb/test/test-base.js:16]: these changes seem nice (altho the eslint override comment is still needed)

**Review by ljharb — CHANGES_REQUESTED** [review]
There's nothing inherently cleaner about defining something in one line versus two.

I commented on the changes that seem good; the rest don't add anything imo.

File: packages/eslint-config-airbnb/test/test-base.js (hunk 1 above)
Comment (mertcanaltin) [reply packages/eslint-config-airbnb/test/test-base.js:16]: I edited @ljharb  thank you 🚀

File: packages/eslint-config-airbnb/test/test-base.js [hunk 2]
```diff
@@ -6,18 +6,17 @@ const base = require('../base');
...
```
Comment (ljharb) [inline packages/eslint-config-airbnb/test/test-base.js:16]:
```suggestion
  files[name] = require(path.join(rulesDir, name)); // eslint-disable-line global-require
```

File: packages/eslint-config-airbnb/test/test-base.js [hunk 3]
```diff
@@ -31,4 +30,9 @@ Object.keys(files).forEach((name) => {
       .filter((ruleId) => ruleId.indexOf('react/') === 0);
     t.deepEquals(reactRuleIds, [], 'there are no react/ rules');
   });
-});
+};
+
+Object.keys(files).forEach((name) => {
+  const config = files[name];
+  checkRules(name, config);
+});
```
Comment (ljharb) [inline packages/eslint-config-airbnb/test/test-base.js:38]: let's revert this

File: packages/eslint-config-airbnb/test/test-base.js [hunk 4]
```diff
@@ -6,18 +6,17 @@ const base = require('../base');
...
 
   // eslint-disable-next-line import/no-dynamic-require
-  files[name] = require(`../rules/${name}`); // eslint-disable-line global-require
+  files[name] = require(path.join(rulesDir, name));
 });
 
-Object.keys(files).forEach((name) => {
-  const config = files[name];
-
+const checkRules = (name, config) => {
```
Comment (ljharb) [inline packages/eslint-config-airbnb/test/test-base.js:19]: and this.

**Review by ljharb — CHANGES_REQUESTED** [review]

File: packages/eslint-config-airbnb/test/test-base.js (hunk 4 above)
Comment (mertcanaltin) [reply packages/eslint-config-airbnb/test/test-base.js]: i got it back dude @ljharb

**Review by ljharb — APPROVED** [review]
Reverted the rest of the changes, re-added the eslint override, and made the same changes in the other test-base file.

**Pull request merged**